server/static/js/tegola.js

```diff
@@ -69,10 +69,24 @@ var app = new Vue({
 				var layers = maps[i].layers;
 				//	iterate our map layers
 				for (var j=0, l=layers.length; j<l; j++){
-					if ((typeof this.map.getLayer(layers[j].name) !== "undefined")) {
-						var color = this.map.getPaintProperty(layers[j].name, 'line-color') ||
-							this.map.getPaintProperty(layers[j].name, 'fill-outline-color') ||
-							this.map.getPaintProperty(layers[j].name, 'circle-color');
+					console.log(layers[j].name);
+					if ((typeof this.map.getLayer(layers[j].name) !== 'undefined')) {
+
+						// these try/catch blocks are because mapbox gl throws an error
+						// if we try to ask for a paint property that does not apply to the feature type
+						// https://github.com/mapbox/mapbox-gl-js/issues/6033
+						var color;
+						try {
+							color = this.map.getPaintProperty(layers[j].name, 'line-color');
+						} catch(e){}
+
+						try {
+							color = this.map.getPaintProperty(layers[j].name, 'fill-outline-color');
+						} catch(e){}
+
+						try {
+							color = this.map.getPaintProperty(layers[j].name, 'circle-color');
+						} catch(e){}
```

Comment:
What is removed is not logically the same as what got added. In the added code the color variable will attempt to be set all three times. Doesn't that make it so that only the last getPoaintPropterty is applied?

Reply:
Another possible solution:
```js
var properties = ['line-color','fill-outline-color','circle-color']
properties.Some(func(p){
      try{
          color = this.map.getPaintProperty(layers[j].name, p);
          return true;
     } catch(e){}
     return false;
})
```

Reply:
Thanks for the snippet. I pushed up something similar.